wave-fe/src/app/round.service.ts

```diff
@@ -16,29 +18,64 @@ export class RoundService {
   private privateAddr = `roundsPrivate/${this.roundId}`;
   private roundRef = firebase.database().ref(this.roundAddr);
   private roundSubject = new ReplaySubject<any>(1);
-  private newRoundData: any = null;
   private score: any = null;
+  private newRoundData: {
+    start: string,
+    startColor: string,
+    end: string,
+    endColor: string,
+    value: number,
+  } = null;
+  private colors = [
+    'blue',
+    'green',
+    'indigo',
+    'orange',
+    'purple',
+    'red',
+    'teal',
+  ];
 
   round$ = this.roundSubject.asObservable();
 
-  constructor(private auth: AuthService, private guessServ: GuessService) {
+  isGuestPlayer$ = this.round$.pipe(
+    map(r => r.phase === 1 && !r.amTeller)
+  )
 
+  constructor(private auth: AuthService, private guessService: GuessService) {
     this.roundRef.on('value', (snapshot) => {
       const roundData = snapshot.val();
-      console.log(roundData, 'roundd ata');
+      console.log(roundData, 'round data');
       this.roundSubject.next(
         this.procRoundData(roundData)
       );
     });
   }
 
+  private generateRandData() {
```

Comment:
Should it be `private generateRandData(): Object`? or should we create an Interface for this Object? @diego0020

Reply:
interface would be the best

Reply:
I've Added a interface, please review latest version :)